modules/apps/forms-and-workflow/calendar/calendar-service/src/main/java/com/liferay/calendar/model/impl/CalendarBookingImpl.java

```diff
@@ -130,7 +130,9 @@ public boolean isMasterRecurringBooking() {
 
 	@Override
 	public boolean isRecurring() {
-		if (Validator.isNotNull(getRecurrence())) {
+		if (Validator.isNotNull(getRecurrence()) ||
+			(getCalendarBookingId() != getRecurringCalendarBookingId())) {
+
```

Comment:
Given the brittle nature of Calendar codebase, I was afraid of changing it. However, it caused no regressions in my tests, so I think we can proceed.

Thanks!